charts/gpu-metrics-exporter/templates/daemonset.yaml

```diff
@@ -47,12 +47,17 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if eq .Values.provider "gke"}}
+      {{- if .Values.gpuMetricsExporter.affinity }}
```

Comment:
so idea here is to introduce generic affinity rule while still keep support for doing the same from `gke` section. Maybe let's add some comment just to use this `gpuMetricsExporter.affinity` as preferable approach, so in future we could maybe unify it and remove `provider.gke`, wdyt?